File: redsheet-portal/src/app/database-models/culture-report-information.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class CultureReportInformation {
    constructor(cultureReportInformation?: CultureReportInformation )  {
        let properties = [ "cultureInformationDetail", "cultureInformationId", "ourCulture", "theirCulture", "cultureSummaryId",];
        for (let property of properties)
            if (cultureReportInformation && cultureReportInformation[property])
                this[property] = cultureReportInformation[property];
    }
 
	cultureInformationDetail : string =   undefined;
 
	cultureInformationId : number =   0 ;
 
	ourCulture : boolean = false ;
 
	theirCulture : boolean = false ;
 
    @range(0,2147483647)
	cultureSummaryId : number =   undefined;


}

File: RedSheet.DbEntities/Models/CultureReportInformation.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("CultureReportInformations", Schema = "dbo")]
	[Serializable]
	public partial class CultureReportInformation  
	{

	public string CultureInformationDetail { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int CultureInformationId { get; set; } 
 
	[Required]
	public bool OurCulture { get; set; } 
 
	[Required]
	public bool TheirCulture { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public CultureSummary CultureSummaryId { get; set; } 

	public CultureReportInformation()
	{
    }

    }
}

